### PR TITLE
Add support for patching Nginx and update Nginx helm chart to 2.0.3

### PR DESCRIPTION
**Pull request by bobdonat**
This pull request:

1. Adds support for patching ingress-nginx.
2. Updates the ingress-nginx helm chart from 1.27.0 to 2.0.3.  Previously, the wrong helm chart was being using for the app version 0.32.0 we have.  This change required changes to the install scripts due to changes in helm chart values, changes in defaults for helm chart values, and changes in names of k8s resources.

### CODE REVIEW

File: thirdparty/charts/ingress-nginx/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
-appVersion: 0.26.1
-description: An nginx Ingress controller that uses ConfigMap to store the nginx configuration.
-engine: gotpl
+appVersion: 0.32.0
+description: Ingress controller for Kubernetes using NGINX as a reverse proxy and
```
Comment (pfmackin) [inline thirdparty/charts/ingress-nginx/Chart.yaml:3]: can you make sure NGINX is the official name

File: thirdparty/charts/ingress-nginx/OWNERS [hunk 2]
```diff
@@ -0,0 +1,5 @@
+approvers:
```
Comment (pfmackin) [inline thirdparty/charts/ingress-nginx/OWNERS:1]: I am assuming that the new ingress-nginx chart is exactly what was pull from the repo (no changes by us)

File: install-olcne.md [hunk 3]
```diff
@@ -240,8 +240,8 @@ Specific steps will differ for each load balancer provider, but a generic config
 |--------------|-----------------------------------------------|------|-------------------------|-------------|
 | Application  | `istio-ingressgateway`                        | TCP  | 80                      | 31380       |
 | Application  | `istio-ingressgateway`                        | TCP  | 443                     | 31390       |
-| Management   | `ingress-controller-nginx-ingress-controller` | TCP  | 80                      | 30080       |
-| Management   | `ingress-controller-nginx-ingress-controller` | TCP  | 443                     | 30443       |
+| Management   | `ingress-controller-ingress-nginx-controller` | TCP  | 80                      | 30080       |
```
Comment (pfmackin) [inline install-olcne.md:243]: I didn't check but i assume this is the exact service name that was incorrectly documented and now you are fixing

File: operator/internal/component/nginx.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+// Verify that Nginx implements Component
+var _ Component = Nginx{}
+
+// Name returns the component name
+func (v Nginx) Name() string {
+	return "ingress-nginx"
+}
+
+// Upgrade upgrades the NGINX ingress controller.
```
Comment (pfmackin) [inline operator/internal/component/nginx.go:28]: can you check the official NGINX name

File: thirdparty/charts/ingress-nginx/OWNERS (hunk 2 above)
Comment (bobdonat) [reply thirdparty/charts/ingress-nginx/OWNERS:1]: yes, no changes from us.  I deleted the old NGINX helm chart and replaced with the 2.0.3 helm chart.

File: install-olcne.md (hunk 3 above)
Comment (bobdonat) [reply install-olcne.md:243]: The service name changed with helm chart 2.0.3 so this change fixes the doc to reflect the service name change.

File: thirdparty/charts/ingress-nginx/Chart.yaml (hunk 1 above)
Comment (bobdonat) [reply thirdparty/charts/ingress-nginx/Chart.yaml:3]: the is the unchanged 2.0.3 Chart.yaml . That is the name they use.

File: operator/internal/component/nginx.go (hunk 4 above)
Comment (bobdonat) [reply operator/internal/component/nginx.go:28]: Looks like NGINX is the official name
https://github.com/kubernetes/ingress-nginx

Comment (bobdonat) [reply operator/internal/component/nginx.go:28]: There a many places in the install scripts that are using Nginx.

**Review by pfmackin — APPROVED** [review]

**Pull request merged**